### PR TITLE
updated submission

### PR DESCRIPTION
**Pull request by ramvenkat98**
Model Hyperparameters:
* On-Policy GRPO
* Learning Rate 5e-5 - did introduce more instability; 3e-5 was more stable but overfit and converged at a smaller validation accuracy
* Kept Std Normalization
* Increased Rollouts + Batch Size (Group Size 16, Train Batch Size = Rollout Batch Size = 512)

Data Sampling Logic:
* Curriculum based on weighting each sample as follows:
    * add 10 to the weight if never seen before
    * add (1 - p(correct)) to the weight where p(correct) is the number of correct rollouts over that prompt’s prior rollouts
    * add 0.25 uniformly

The curriculum is in the final run but it seems like it would likely be because it forces the model to see new examples early on and discourages it from overfitting by seeing the same examples quickly. The (1 - p(correct)) part alone doesn’t seem to do much, and we could reproduce most of the gain from the curriculum by just iterating through the samples in order of least seen to most seen (instead of doing random resampling each time).

Did Not Work:
* Varying batch sizes and learning rates over the course of the training run
    * Up to train step 90, using train batch size = rollout batch size = 256, group size = 8
    * Up to train step 200, increasing batch sizes to 1024 and group size to 32
    * Decreasing learning rate after train step 200

This was removed in the end - the increase in rollout sizes midway through performed comparably to increasing rollout sizes from the start itself (rather than midway), which was done in the end. Decreasing the learning rate hurt rather than helped (which makes sense because the model is not constrained in terms of under-fitting the prompts, it tends to overfit and achieve very high accuracy on the train prompts.)

### CODE REVIEW

**Comment by marcelroed** [comment]
Sorry, we got a commit that created merged conflicts for all -- can you rebase/merge with the current version (and space it out nicely)?
